boards/DIM/Test/Src/seven_seg_displays_test.cpp

```diff
@@ -0,0 +1,128 @@
+#include "fff.h"
+#include "gtest/gtest.h"
+
+extern "C"
+{
+#include "App_SevenSegDisplay.h"
+#include "App_SevenSegDisplays.h"
+
+    DEFINE_FFF_GLOBALS;
+    FAKE_VALUE_FUNC(float, get_state_of_charge);
+    FAKE_VOID_FUNC(set_right_hex_digit, uint8_t);
+    FAKE_VOID_FUNC(set_middle_hex_digit, uint8_t);
+    FAKE_VOID_FUNC(set_left_hex_digit, uint8_t);
+}
+
+class SevenSegDisplaysTest : public testing::Test
+{
+  protected:
+    void SetUp() override
+    {
+        right_display  = App_SevenSegDisplay_Create(set_right_hex_digit);
+        middle_display = App_SevenSegDisplay_Create(set_middle_hex_digit);
+        left_display   = App_SevenSegDisplay_Create(set_left_hex_digit);
+
+        seven_segment_displays = App_SevenSegDisplays_Create(
+            get_state_of_charge, left_display, middle_display, right_display);
+
+        RESET_FAKE(get_state_of_charge);
+        RESET_FAKE(set_right_hex_digit);
+        RESET_FAKE(set_middle_hex_digit);
+        RESET_FAKE(set_left_hex_digit);
+
+        FFF_RESET_HISTORY();
+    }
+    void TearDown() override
+    {
+        App_SevenSegDisplay_Destroy(right_display);
+        App_SevenSegDisplay_Destroy(middle_display);
+        App_SevenSegDisplay_Destroy(left_display);
+        App_SevenSegDisplays_Destroy(seven_segment_displays);
+    }
+
+    struct SevenSegDisplay * left_display;
+    struct SevenSegDisplay * middle_display;
+    struct SevenSegDisplay * right_display;
+    struct SevenSegDisplays *seven_segment_displays;
+};
+
+TEST_F(SevenSegDisplaysTest, write_max_state_of_chrage)
+{
+    get_state_of_charge_fake.return_val = 100.0f;
+
+    App_SevenSegDisplays_WriteStateOfCharge(seven_segment_displays);
+
+    ASSERT_EQ(1, set_left_hex_digit_fake.arg0_history[0]);
+    ASSERT_EQ(0, set_middle_hex_digit_fake.arg0_history[0]);
+    ASSERT_EQ(0, set_right_hex_digit_fake.arg0_history[0]);
+
+    ASSERT_EQ(1, get_state_of_charge_fake.call_count);
+    ASSERT_EQ(1, set_left_hex_digit_fake.call_count);
+    ASSERT_EQ(1, set_middle_hex_digit_fake.call_count);
+    ASSERT_EQ(1, set_right_hex_digit_fake.call_count);
+
+    ASSERT_EQ(fff.call_history[0], (void *)get_state_of_charge);
```

Comment:
Wasn't sure if the `call_history` is excessive or not :thinking:

Reply:
I think it's probably excessive. Let's consider what we really want to check here. We just want to check that the `left` `middle` and `right` setters were called with the values that we expect. The rest of the stuff we're checking here could change and we'd still consider the function "correct", which is a good indicator that we shouldn't be checking it.

Reply:
Yeah I think you are right about this one. Removing all things call history related.